packages/contracts/contracts/LenderManager.sol

```diff
@@ -79,6 +89,110 @@ contract LenderManager is
     }
 
     function _baseURI() internal view override returns (string memory) {
-        return "";
+        return "data:image/svg+xml;charset=utf-8,";
+    }
+
+    struct LoanInformation {
+
+        address principalTokenAddress;
+        uint256 principalAmount;
+        uint16 interestRate;
+        uint32 loanDuration;
+
+
+    }
+
+
+    function _getLoanInformation(uint256 tokenId) 
+    internal view 
+    returns (LoanInformation memory loanInformation_) {
+
+        Bid memory bid = ITellerV2Storage(owner()).bids(tokenId);
+        
+        LoanInformation memory loanInformation = LoanInformation({
+            principalTokenAddress: address(bid.loanDetails.lendingToken),
+            principalAmount: bid.loanDetails.principal,
+            interestRate: bid.terms.APR,
+            loanDuration: bid.loanDetails.loanDuration
+        });
+
+
+        return loanInformation_;
```

Comment:
you are not returning the correct variable - you instantiate a new variable on line 112 instead of using the variable used in the return statement on line 108. you should also delete line 120